src/java/org/apache/cassandra/db/tries/CollectionMergeCursor.java

```diff
@@ -107,16 +109,19 @@ <I> CollectionMergeCursor(Trie.CollectionMergeResolver<T> resolver, Direction di
         for (I src : inputs)
         {
             C cursor = extractor.apply(src, direction);
-            assert cursor.depth() == 0;
             if (i >= 0)
                 heap[i] = cursor;
             else
                 head = cursor;
             ++i;
         }
         // The cursors are all currently positioned on the root and thus in valid heap order.
+        assert Arrays.stream(heap).allMatch(x -> equalCursor(head, x));
     }
 
+    // TODO: consider if making these return boolean would be an optimization for some calls
+    // TODO: consider if head call can be included in these
```

Comment:
Should these TODOs be done in this PR ?

Reply:
Applied one and dropped the boolean return as it's not that helpful.